netket/graph/lattice.py

```diff
@@ -534,10 +499,101 @@ def position_from_basis_coords(self, basis_coords: CoordT) -> PositionT:
         Throws a KeyError if no site is found for any of the coordinates.
         """
         ids = self.id_from_basis_coords(basis_coords)
-        if _np.any(ids == None):  # pylint: disable=singleton-comparison
-            raise KeyError(f"No site found at at least one of {basis_coords}")
         return self.positions[ids]
 
+    def to_reciprocal_lattice(self, ks: Array) -> Array:
+        """
+        Takes wave vectors in Cartesian axes and converts them into the reciprocal
+        basis **of the simulation box.**
+        Valid wave vector components in this basis are integers in (periodic BCs)
+        or zero (in open BCs).
+        Throws a `ValueError` if any of the supplied wave vectors are not
+        reciprocal lattice vectors of the simulation box.
+        """
+        # Ensure that ks has at least 2 dimensions
+        ks = _np.asarray(ks)
+        if ks.ndim == 1:
+            ks = ks[_np.newaxis, :]
+
+        result = ks @ self._lattice_dims.T / (2 * pi)
+        # Check that these are integers
+        is_valid = is_approx_int(result)
+        if not _np.all(is_valid):
+            raise self.InvalidWaveVectorError(
+                "Some wave vectors are not reciprocal lattice vectors of the simulation box"
+            )
+
+        result = _np.asarray(_np.rint(result), dtype=int)
+        # For axes with non-periodic BCs, the k-component must be 0
+        is_valid = _np.logical_or(self.pbc, result == 0)
+        if not _np.all(is_valid):
+            raise self.InvalidWaveVectorError(
+                "Some wave vectors are inconisistent with open boundary conditions"
+            )
+
+        return result
+
+    # Generating space groups
+    # -----------------------------------------------------------------------
+    def space_group_builder(
+        self, point_group: Optional[PointGroup] = None
+    ) -> "SpaceGroupBuilder":
+        """
+        Returns a `SpaceGroupBuilder` object that represents the space group
+        generated by the translation symmetries of `self` and the elements of
+        `point_group`. This object can be used to produce `PermutationGroup`s
+        that describe the action of these groups on the sites of `self`, and to
+        generate space group irreps for symmetrising wave functions.
+        If no `point_group` is specified, uses the point group provided upon
+        construction.
+        """
```

Comment:
missing arg and returns from docstring.

Also, docstring should be a short description (the recap) that gets picked up in some previews of the docs, newline, then a long description.